src/pymor/reductors/neural_network.py

```diff
@@ -463,20 +555,54 @@ def compute_training_data(self):
         """Compute the training samples (the outputs to the parameters of the training set)."""
         with self.logger.block('Computing training samples ...'):
             self.training_data = []
-            for datum in self.training_set:
+            for i, mu in enumerate(self.training_set):
+                if not self.fom:
+                    samples = self._compute_sample(mu, output=self.training_outputs[i*self.nt:(i+1)*self.nt])
+                else:
+                    samples = self._compute_sample(mu)
+                for sample in samples:
+                    self._update_scaling_parameters(sample)
+                self.training_data.extend(samples)
+
+    def compute_validation_data(self):
+        """Compute the training samples (the outputs to the parameters of the training set)."""
```

Comment:
```suggestion
        """Compute the validation samples (the outputs to the parameters of the training set)."""
```
Here, not only the outputs are computed, but actually the pairs of parameters and outputs, right? Maybe one could slightly adjust the docstring to clarify this.